Marlin/src/module/tool_change.cpp

```diff
@@ -681,7 +681,7 @@ void tool_change(const uint8_t tmp_extruder, const float fr_mm_s/*=0.0*/, bool n
       enable_solenoid_on_active_extruder();
     #endif
 
-    #if HAS_SOFTWARE_ENDSTOPS && ENABLED(DUAL_X_CARRIAGE) && HAS_WORKSPACE_OFFSET
+    #if HAS_SOFTWARE_ENDSTOPS && ENABLED(DUAL_X_CARRIAGE)
       update_software_endstops(X_AXIS);
     #endif
```

Comment:
There's no reason to call `update_software_endstops` if the endstops aren't going to change. The soft endstops only change on a tool-change if using something like a Dual X Carriage or a Switching Nozzle, whose X limits may be different depending on which carriage(s) or hotend is/are engaged. Examine `update_software_endstops` to determine the exact cases where the soft endstops could, in fact, be modified.